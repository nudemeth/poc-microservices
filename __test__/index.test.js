import React from 'react';
import Router from 'next/router'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Index from '../pages/index';

const mockedRouter = { prefetch: () => {} }
Router.router = mockedRouter

describe('Index page', () => {
    it('Should show "This is Index Page: ABC" in Index page', () => {
        const index = shallow(<Index greeting="ABC" />);
        expect(index.find('h1').text()).toEqual('This is Index Page: ABC');
    });
});

describe('Index page with Snapshot Testing', () => {
    it('Should show "This is Index Page: ABC" in Index page', () => {
        const component = renderer.create(<Index greeting="ABC" />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});