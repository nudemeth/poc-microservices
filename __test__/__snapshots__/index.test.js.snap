// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page with Snapshot Testing Should show "This is Index Page: ABC" in Index page 1`] = `
<div
  className="Container-root-1"
>
  <div
    className="Container-appFrame-2"
  >
    <header
      className="MuiPaper-root-15 MuiPaper-shadow4-21 MuiAppBar-root-7 MuiAppBar-positionFixed-8 MuiAppBar-colorPrimary-13 mui-fixed Header-appBar-6"
    >
      <div
        className="MuiToolbar-root-42 MuiToolbar-gutters-43"
      >
        <button
          aria-label="Menu"
          className="MuiButtonBase-root-50 MuiIconButton-root-44 Header-menuButton-5"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label-49"
          >
            <span
              aria-hidden="true"
              className="material-icons MuiIcon-root-52 MuiIcon-fontSize-58"
            >
              menu
            </span>
          </span>
          <span
            className="MuiTouchRipple-root-59"
          />
        </button>
        <p
          className="MuiTypography-root-66 MuiTypography-body1-75 MuiTypography-colorInherit-85 Header-flex-4"
          type="title"
        >
          Title
        </p>
      </div>
    </header>
    <div
      className="MuiHiddenCss-smDown-98"
    >
      <div
        className="MuiDrawer-docked-108 SideBar-drawerDocked-91"
      >
        <div
          className="MuiPaper-root-15 MuiPaper-shadow0-17 MuiDrawer-paper-109 SideBar-drawerPaper-92 MuiDrawer-paperAnchorLeft-110 MuiDrawer-paperAnchorDockedLeft-114"
        >
          <div
            className="DrawerMenu-drawerHeader-119"
          />
          <hr
            className="MuiDivider-root-120 MuiDivider-default-122"
          />
          <ul
            className="MuiList-root-125 MuiList-padding-126"
          >
            <div
              className="MuiButtonBase-root-50 MuiListItem-root-129 MuiListItem-default-132 MuiListItem-gutters-136 MuiListItem-button-137"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-52"
              >
                home
              </span>
              <div
                className="MuiListItemText-root-139"
              >
                <h3
                  className="MuiTypography-root-66 MuiTypography-subheading-73 MuiListItemText-primary-142"
                >
                  Home
                </h3>
              </div>
              <span
                className="MuiTouchRipple-root-59"
              />
            </div>
            <div
              className="MuiButtonBase-root-50 MuiListItem-root-129 MuiListItem-default-132 MuiListItem-gutters-136 MuiListItem-button-137"
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-52"
              >
                info
              </span>
              <div
                className="MuiListItemText-root-139"
              >
                <h3
                  className="MuiTypography-root-66 MuiTypography-subheading-73 MuiListItemText-primary-142"
                >
                  Info
                </h3>
              </div>
              <span
                className="MuiTouchRipple-root-59"
              />
            </div>
          </ul>
        </div>
      </div>
    </div>
    <main
      className="Container-content-3"
      role="main"
    >
      <h1>
        This is Index Page: 
        ABC
      </h1>
    </main>
  </div>
</div>
`;
